modules/api/src/main/java/org/eclipse/xpanse/api/controllers/IsvCloudCredentialsApi.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.api.controllers;
+
+import static org.eclipse.xpanse.modules.models.security.constant.RoleConstants.ROLE_ISV;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.credential.CredentialCenter;
+import org.eclipse.xpanse.modules.models.credential.AbstractCredentialInfo;
+import org.eclipse.xpanse.modules.models.credential.CreateCredential;
+import org.eclipse.xpanse.modules.models.credential.enums.CredentialType;
+import org.eclipse.xpanse.modules.models.service.common.enums.Csp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * REST interface method for managing cloud provider credentials for users of a role ISV.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse/isv")
+@CrossOrigin
+@Secured({ROLE_ISV})
+public class IsvCloudCredentialsApi {
+
+    private final CredentialCenter credentialCenter;
+
+    @Autowired
+    public IsvCloudCredentialsApi(CredentialCenter credentialCenter) {
+        this.credentialCenter = credentialCenter;
+    }
+
+    /**
+     * Users in the ISV role get the list of available credential types of the cloud service
+     * provider.
+     *
+     * @param csp The cloud service provider.
+     * @return Returns list the available credential types of the cloud service provider.
+     */
+    @Tag(name = "Credentials Management",
+            description = "APIs for managing isv's cloud provider credentials")
+    @GetMapping(value = "/credential_types",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "Users in the ISV role get the list of available credential types of"
+            + " the cloud service provider.")
+    public List<CredentialType> getIsvCloudCredentialTypes(
```

Comment:
This is a point that I missed to mention. This method and the next method (getIsvCloudCredentialCapabilities) are common to both ISV and end users. 
We can move it to another common API controller - CredentialsConfigApi. The definition of cloud credentials will be at just one place and will be used by both ISV and end users.